simulator/assembly/__tests__/singleton-fail.ts

```diff
@@ -0,0 +1,45 @@
+import { logging, PersistentSet, Context } from "near-sdk-as";
+
+@nearBindgen
+export class Singleton {
+
+  private visitors: PersistentSet<string> = new PersistentSet("v");
+  private _owner: string
+  private last_visited: string = "NULL";
+  constructor(owner: string){
+    this._owner = owner;
+  };
+
+  owner(): string {
+    return this._owner;
+  }
+
+  @change()
+  visit(): void {
+    if (!this.visitors.has(Context.sender)) {
+      logging.log("Visited by " + Context.sender);
+      this.visitors.add(Context.sender);
+    }
+    this.last_visited = Context.sender;
+  }
+
+  visit_without_change(): void {
+    this.visit();
+  }
+
+  hasVisited(visitor: string): bool {
+    return this.visitors.has(visitor);
+  }
+
+  lastVisited(): string {
+    return this.last_visited || "";
+  }
+
+  private hasNotVisited(visitor: string): boolean {
+    return !this.hasNotVisited(visitor);
+  }
```

Comment:
Are any of these methods used? Seems like this file is only here to test that the `new` function below causes errors. If that's true, let's get rid of everything except the code under test.

Reply:
This method shouldn't end up on the final binary's exports.  I didn't have a test for this just checked  the output myself.  I'll add a test.

Reply:
nevermind I do have a test for this:

```ts
  it("should not have private methods", () => {
    init()
    let res = alice.call_other("singleton", "hasNotVisited", {});
    expect(res.err["FunctionCallError"]["MethodResolveError"]).toContain("MethodNotFound");
  })
```

Reply:
Right, but that uses a different file. Confusing, huh? 😝